lib/input_event.ex

```diff
@@ -16,9 +16,22 @@ defmodule InputEvent do
   @doc """
   Start a GenServer that reports events from the specified input event device
   """
-  @spec start_link(Path.t()) :: :ignore | {:error, any()} | {:ok, pid()}
-  def start_link(path) do
-    GenServer.start_link(__MODULE__, [path, self()])
+  @spec start_link(Path.t(), grab: boolean(), callback: GenServer.server()) ::
+          :ignore | {:error, any()} | {:ok, pid()}
+  def start_link(path, opts \\ [grab: false]) do
```

Comment:
NOTE: After writing the comment below I totally understand why you combined the commits. I hope you see this before breaking them apart! So sorry for not understanding before!!!

I have some concern about 2 argument `start_link` functions since they're somewhat unusual in Elixir. How about we make the switch to passing a keyword list. Then we won't need the `child_spec` functions.

This change needs some documentation updates and a change to the `init/1` callback, but hopefully it gives you an idea of what I'm thinking. 

```suggestion
  @spec start_link(Path.t() | keyword()) :: GenServer.on_start()
  def start_link(path) when is_binary(path) do
    start_link(path: path)
  end
  
  def start_link(init_args) when is_list(init_args) do
    init_args[:path] || raise ArgumentError, "InputEvent requires a input event device path"
    
    updated_args = Keyword.put_new(init_args, :callback, self())
    GenServer.start_link(__MODULE__, updated_args)
  end
```